### ISSUE TITLE
Disabling logging is not working

### ISSUE THREAD

**Issue by yo-ga** [pycontribs/jira#1009]
**Describe the bug**
In [the document](https://jira.readthedocs.io/en/latest/api.html#jira.JIRA), there is a parameter named ```logging```.
The comment in the ```__init__``` says "Determine whether or not logging should be enabled. (Default: True)".
If the programmer tries to disable the log, it doesn't work.

**To Reproduce**
```py
from jira import JIRA
jira_client = JIRA("https://UNABLE_CONNECT_URL", auth=("username", "password"), logging=False)
```
Output:
```
WARNING:root:HTTPSConnectionPool(host='unable_connect_url', port=443): Max retries exceeded with url: /rest/auth/1/session (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x10ae850b8>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known',)) while doing POST https://UNABLE_CONNECT_URL/rest/auth/1/session [{'data': '{"username": "username", "password": "password"}', 'headers': {'User-Agent': 'python-requests/2.24.0', 'Accept-Encoding': 'gzip, deflate', 'Accept': 'application/json,*.*;q=0.9', 'Connection': 'keep-alive', 'Cache-Control': 'no-cache', 'Content-Type': 'application/json', 'X-Atlassian-Token': 'no-check'}}]
WARNING:root:Got ConnectionError [HTTPSConnectionPool(host='unable_connect_url', port=443): Max retries exceeded with url: /rest/auth/1/session (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x10ae850b8>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known',))] errno:None on POST https://UNABLE_CONNECT_URL/rest/auth/1/session
{'response': None, 'request': <PreparedRequest [POST]>}\{'response': None, 'request': <PreparedRequest [POST]>}
WARNING:root:Got recoverable error from POST https://UNABLE_CONNECT_URL/rest/auth/1/session, will retry [1/3] in 8.383788695379353s. Err: HTTPSConnectionPool(host='unable_connect_url', port=443): Max retries exceeded with url: /rest/auth/1/session (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x10ae850b8>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known',))
WARNING:root:HTTPSConnectionPool(host='unable_connect_url', port=443): Max retries exceeded with url: /rest/auth/1/session (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x10ae85400>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known',)) while doing POST https://UNABLE_CONNECT_URL/rest/auth/1/session [{'data': '{"username": "username", "password": "password"}', 'headers': {'User-Agent': 'python-requests/2.24.0', 'Accept-Encoding': 'gzip, deflate', 'Accept': 'application/json,*.*;q=0.9', 'Connection': 'keep-alive', 'Cache-Control': 'no-cache', 'Content-Type': 'application/json', 'X-Atlassian-Token': 'no-check'}}]
WARNING:root:Got ConnectionError [HTTPSConnectionPool(host='unable_connect_url', port=443): Max retries exceeded with url: /rest/auth/1/session (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x10ae85400>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known',))] errno:None on POST https://UNABLE_CONNECT_URL/rest/auth/1/session
{'response': None, 'request': <PreparedRequest [POST]>}\{'response': None, 'request': <PreparedRequest [POST]>}
WARNING:root:Got recoverable error from POST https://UNABLE_CONNECT_URL/rest/auth/1/session, will retry [2/3] in 16.6335029520674s. Err: HTTPSConnectionPool(host='unable_connect_url', port=443): Max retries exceeded with url: /rest/auth/1/session (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x10ae85400>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known',))
WARNING:root:HTTPSConnectionPool(host='unable_connect_url', port=443): Max retries exceeded with url: /rest/auth/1/session (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x10ae71f60>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known',)) while doing POST https://UNABLE_CONNECT_URL/rest/auth/1/session [{'data': '{"username": "username", "password": "password"}', 'headers': {'User-Agent': 'python-requests/2.24.0', 'Accept-Encoding': 'gzip, deflate', 'Accept': 'application/json,*.*;q=0.9', 'Connection': 'keep-alive', 'Cache-Control': 'no-cache', 'Content-Type': 'application/json', 'X-Atlassian-Token': 'no-check'}}]
WARNING:root:Got ConnectionError [HTTPSConnectionPool(host='unable_connect_url', port=443): Max retries exceeded with url: /rest/auth/1/session (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x10ae71f60>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known',))] errno:None on POST https://UNABLE_CONNECT_URL/rest/auth/1/session
{'response': None, 'request': <PreparedRequest [POST]>}\{'response': None, 'request': <PreparedRequest [POST]>}
WARNING:root:Got recoverable error from POST https://UNABLE_CONNECT_URL/rest/auth/1/session, will retry [3/3] in 38.93275918765591s. Err: HTTPSConnectionPool(host='unable_connect_url', port=443): Max retries exceeded with url: /rest/auth/1/session (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x10ae71f60>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known',))

ConnectionError: HTTPSConnectionPool(host='unable_connect_url', port=443): Max retries exceeded with url: /rest/auth/1/session (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x10ae85470>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known',))
```

**Expected behavior**
Only print the connection error message.

**Version Information**
Python Interpreter: 3.6.7
jira-python: 2.0.0
OS: MacOS Catalina 10.15.6
IPython (Optional): 7.13.0
Other Dependencies:
- requests: 2.24.0
- requests-oauthlib: 1.3.0
- requests-toolbelt: 0.9.1

**Additional context**
For now, I use ```logging.disable()``` to disable logging.
```py
import logging
from jira import JIRA

logging.disable(logging.CRITICAL)
jira_client = JIRA("https://UNABLE_CONNECT_URL", auth=("username", "password"))
logging.disable(logging.NOSET)
```

**Comment by maskeeter** [comment]
this might not be the same problem but a related one: The Jira logger is actually logging to the root logger [not to 'jira' logger](https://github.com/pycontribs/jira/blob/7fa3a454c08a14e2d7d7670fcfa87482e16936ba/jira/client.py#L95), while bool logging parameter is [not used at all](https://github.com/pycontribs/jira/blob/7fa3a454c08a14e2d7d7670fcfa87482e16936ba/jira/client.py#L455).

**Comment by asqui** [comment]
There was a PR relating to this some years back, but it never got merged: #329

The branch and/or PR comments may be useful for someone who wants to prepare a new PR with suggested changes to address the problems described in this issue.
